src/fu-engine.c

```diff
@@ -563,37 +584,65 @@ fu_engine_verify_update (FuEngine *self, const gchar *device_id, GError **error)
 		return FALSE;
 	}
 
-	/* load existing store */
-	store = fu_engine_load_verify_store (error);
-	if (store == NULL)
+	/* build XML */
+	component = xb_builder_node_insert (NULL, "component",
+					    "type", "firmware",
+					    NULL);
+	provides = xb_builder_node_insert (component, "provides", NULL);
+	guids = fu_device_get_guids (device);
+	for (guint i = 0; i < guids->len; i++) {
+		const gchar *guid = g_ptr_array_index (guids, i);
+		g_autoptr(XbBuilderNode) provide = NULL;
+		provide = xb_builder_node_insert (provides, "firmware",
+						  "type", "flashed",
+						  NULL);
+		xb_builder_node_set_text (provide, guid, -1);
+	}
+	releases = xb_builder_node_insert (component, "releases", NULL);
+	release = xb_builder_node_insert (releases, "release",
+					  "version", fu_device_get_version (device),
+					  NULL);
+	for (guint i = 0; i < checksums->len; i++) {
+		const gchar *checksum = g_ptr_array_index (checksums, i);
+		GChecksumType kind = fwupd_checksum_guess_kind (checksum);
+		g_autoptr(XbBuilderNode) csum = NULL;
+		csum = xb_builder_node_insert (provides, "checksum",
```

Comment:
Don't you want to check the results of `_g_checksum_type_to_string` for `NULL` before filling this in (make the node fail if it's an unknown checksum).

Reply:
I don't think `fwupd_checksum_guess_kind()` can return an invalid checksum, it falls back to assuming SHA1.

Reply:
Then I think you should also configure `_g_checksum_type_to_string` to fallback to `sha1` in the same instance.

Reply:
Done, agreed.